python/buckets.cpp

```diff
@@ -68,9 +89,52 @@ template <class T> auto get_buffer(py::object &obj) {
 } // namespace
 
 void init_buckets(py::module &m) {
-  bind_buckets<Variable>(m);
-  bind_buckets<DataArray>(m);
-  bind_buckets<Dataset>(m);
+  bind_bins<Variable>(m);
+  bind_bins<DataArray>(m);
+  bind_bins<Dataset>(m);
+
+  bind_bin_size<Variable>(m);
+  bind_bin_size<DataArray>(m);
+  bind_bin_size<Dataset>(m);
+
+  m.def("is_bins", variable::is_buckets);
+  m.def("is_bins", [](const DataArrayConstView &array) {
+    return dataset::is_buckets(array);
+  });
+  m.def("is_bins", [](const DatasetConstView &dataset) {
+    return dataset::is_buckets(dataset);
+  });
+
+  m.def("bins_begin_end", [](const VariableConstView &var) -> py::object {
+    if (var.dtype() == dtype<bucket<Variable>>)
```

Comment:
Not so straightforwardly, because the operator is templated, so it cannot passed directly as a template arg. We'd need something like `callDType`, but that requires wrapping everything in helper classes. We can do so once more than two cases are needed, or wait until there may be a complete change to the mechanism.